Add explicit arguments for Vertex AI eval clients

taniokay opened the pull request:
This PR changes the Vertex AI eval clients to pass necessary info from EvalClient init arguments instead of environment variables so that we can easily use Vertex AI client and non Vertex AI client.
This also bumps the version to `0.10.0.dev3`.

```python
from dotenv import load_dotenv

load_dotenv()

from langcheck.metrics.eval_clients import (
    AnthropicEvalClient,
    GeminiEvalClient,
)

# The following environment variables should be set
# ANTHROPIC_VERTEX_PROJECT_ID=<your-project-id>
# CLOUD_ML_REGION=<region>  (e.g. europe-west1)
# GOOGLE_APPLICATION_CREDENTIALS=<path-to-credentials-file>
anthropic_client = AnthropicEvalClient(vertexai=True)

# The following environment variables should be set
# GOOGLE_CLOUD_PROJECT=<your-project-id>
# GOOGLE_CLOUD_LOCATION=<location>  (e.g. europe-west1)
# GOOGLE_APPLICATION_CREDENTIALS=<path-to-credentials-file>
gemini_client = GeminiEvalClient(
    model_name="gemini-2.0-flash",
    embed_model_name="text-embedding-004",
    vertexai=True
)

anthropic_response = anthropic_client.get_text_responses(
    prompts=["Hello"]
)
gemini_response = gemini_client.get_text_responses(
    prompts=["Hello"]
)

```

yosukehigashi commented on src/langcheck/metrics/eval_clients/_anthropic.py:
```diff
@@ -28,16 +28,16 @@ def __init__(
         system_prompt: str | None = None,
+        project: str | None = None,
+        location: str | None = None,
+        credentials: Credentials | None = None,
```
Maybe rename these vars to be clear that this is how you enable Vertex AI mode? E.g. something like google_credentials or google_vertex_ai_credentials

yosukehigashi commented on src/langcheck/metrics/eval_clients/_anthropic.py:
```diff
@@ -28,16 +28,16 @@ def __init__(
         """
         Initialize the Anthropic evaluation client. The authentication
         information is automatically read from the environment variables,
         so please make sure ANTHROPIC_API_KEY is set.
```
not related to this PR, but `ANTHROPIC_API_KEY` doesn't need to be set when using Vertex, right?

taniokay replied:
Yeah right, maybe we can change the docstring to specify that

yosukehigashi replied:
I think we can (and should) read the API keys from env vars. Most tools do it it like this, and it's easier for the user.

(As long as it doesn't need to be updated ever, I think it's fine to use env vars. And this is the case for API keys - you just set it once and shouldn't need to edit it while your program is running)

taniokay replied:
Sorry for my late suggestion which would make changes again, but what do you think about adding only explicit `vertexai` argument to eval clients and reading other necessary information from env vars? That doesn't require users to instantiate `Credentials` before using VertexAI eval client, and we can use Vertex AI and non Vertex AI eval clients at the same time

taniokay replied:
Added change in [6741071](https://github.com/citadel-ai/langcheck/pull/195/commits/6741071ebb949b7eefb638ac56161909587388e4)

yosukehigashi replied:
nice, this is much cleaner!! yeah I think this is the best approach - it's easy for users, and there are no surprises about how things behave based on the env variables (whereas before, it was hard to tell whether things would run in Vertex AI mode or not)

yosukehigashi reviewed (approved):
Awesome, LGTM! Just a couple nits about comments

yosukehigashi reviewed (changes requested):
Actually one more comment - could we add some unit tests in `test_anthropic.py` and `test_gemini.py`? They can be super simple (i.e. most things can be Mocked)

taniokay commented:
I made some changes:
- I reverted the project_id, location, and credentials arguments change, and instead I added `vertexai`. We can give other info via var envs. https://github.com/citadel-ai/langcheck/pull/195#discussion_r2032597330
- I also reverted the version bump as I'm going to create some PRs [648c2a2](https://github.com/citadel-ai/langcheck/pull/195/commits/648c2a2902d66eaaba6cb88e0c819bfde7718135)
- I added fixture in pytest that cleanup the env vars [9d97c71](https://github.com/citadel-ai/langcheck/pull/195/commits/9d97c719a7cf6a44ff1c3f4b9f475b02390b6220)
- I fixed `chikkarpy >= 0.1.1` [0ae19e6](https://github.com/citadel-ai/langcheck/pull/195/commits/0ae19e68ef78ec022fc486604a2eb90804d5763d)

@yosukehigashi Could you take another look?

yosukehigashi commented on src/langcheck/metrics/eval_clients/_anthropic.py:
```diff
@@ -27,17 +26,18 @@ def __init__(
         Initialize the Anthropic evaluation client. The authentication
         information is automatically read from the environment variables,
         so please make sure ANTHROPIC_API_KEY is set.
         If you want to use Vertex AI, please set the following environment
```
maybe add "and set the `vertexai` parameter to True"?

taniokay commented:
CI tried to install `chikkarpy ==0.1.0` which seems having dependencies issue, but I didn't see this issue in the previous CI failure. I just pinned `chikkarpy>=0.1.1 and it looks resolved.
I'm seeing another CI failure related to HanLP error but it seems from before so I think we can ignore this for now. https://github.com/citadel-ai/langcheck/pull/103#issuecomment-2033495403

yosukehigashi commented:
> CI tried to install `chikkarpy ==0.1.0` which seems having dependencies issue, but I didn't see this issue in the previous CI failure. I just pinned `chikkarpy>=0.1.1 and it looks resolved.

oh that's weird, I wonder why it tried to install an older release.. SG to pin it though!
